kolibri/core/logger/upgrade.py

```diff
@@ -44,3 +48,12 @@ def fix_asymptotic_progress():
         if not quiz:
             # If this isn't a quiz generate the appropriate completion notification for the resource.
             parse_summarylog(summarylog)
+
+
+@version_upgrade(old_version="<0.16.0")
+def fix_duplicated_attempt_logs():
+    """
+    Migrate any AttemptLogs for Quizzes and PracticeQuizzes that have been duplicated - i.e. have the same
+    item and non-null masterylog_id.
+    """
+    consolidate_quiz_attempt_logs(AttemptLog.objects.all())
```

Comment:
I wonder whether KDP will be able to handle this during its upgrade?

Reply:
(at least it shouldn't change most records, just problematic ones, so it won't lead to a massive case of "need to send all logs back down to clients again")

Reply:
I assumed that we would probably blocklist this like we have all the other data migration upgrades.

Reply:
@jredrejo did discuss trying to have a more systematic way to mark these kinds of upgrades for exclusion for KDP, but we did not come up with anything that fit the bill just yet.